AdminUi/src/AdminUi/ClientApp/src/app/components/quotas/assign-quotas-dialog/assign-quotas-dialog.component.ts

```diff
@@ -87,7 +77,12 @@ export class AssignQuotasDialogComponent {
             period: this.period!
         };
 
-        this.dialogRef.close(quota);
+        const callback = this.data.callback;
```

Comment:
Isn't the callback available in the constructor already? Because then I would set it to a strongly typed field there already.

Reply:
Yes it is but.. with that i cannot trigger the parent method in the dialog component. One of the ways is doing like this.. now when I think about it I found an even cleaner solution via EventEmmitter.

Reply:
I don't think that EventEmitter is a good idea. Too much failure potential. If you are on the Identity Details of an Identity, then switch to the Tier Details and don't unsubscribe before that and then assign quota to the Tier, this quota might also end up in the identity.

Reply:
> with that i cannot trigger the parent method in the dialog component

I don't understand why this shouldn't be possible

Reply:
What I wanted to say there when I trigger the parent component in this dialog component then i would have to trigger all other parent components that are calling this dialog in my case now is two of those.

Reply:
Hmm, I still don't understand. Why are there multiple parents? One dialog is always only created by one component.

Reply:
Yes, but the dialog is the same and when you want to access a parent method from a component in a dialog it makes not reusable.. if i call a method from a one component to send the data to that components method i cannot send that data to the other component i would simply need to call that other components method and do the same.